### ISSUE TITLE
Unable to use the bundle with Symfony security

### ISSUE THREAD

**Issue by rkmourya** [IDCI-Consulting/IDCIKeycloakSecurityBundle#4]
In my security.yaml I am having below entry
```security:
    # https://symfony.com/doc/current/security.html#where-do-users-come-from-user-providers
    providers:
        in_memory: { memory: ~ }
    firewalls:
        statesapi:
            pattern: ^/states/
            guard:
                provider: idci_keycloak_bearer_security_provider
                authenticators:
                    - IDCI\Bundle\KeycloakSecurityBundle\Security\Authenticator\KeycloakBearerAuthenticator
```

However this throws me below error

```
Argument 5 passed to Symfony\Component\Security\Http\Firewall\ExceptionListener::__construct() must implement interface Symfony\Component\Security\Http\EntryPoint\AuthenticationEntryPointInterface or be null, instance of IDCI\Bundle\KeycloakSecurityBundle\Security\Authenticator\KeycloakBearerAuthenticator given, called in C:\EF\abc-gateway\symfony\var\cache\dev\ContainerHVUqm4t\getSecurity_Firewall_Map_Context_StatesapiService.php on line 24
```

**Comment by konandrum** [comment]
You seem to secure an API, try this configuration:

```yaml
imports:
    - { resource: '@IDCIKeycloakSecurityBundle/Resources/config/security.yaml' }

security:
    firewalls:
        statesapi:
            pattern: ^/states/
            provider: idci_keycloak_bearer_security_provider
            simple_preauth:
                authenticator: IDCI\Bundle\KeycloakSecurityBundle\Security\Authenticator\KeycloakBearerAuthenticator
```

Don't forget to clear your cache.
Which Symfony version are you using ?

**Comment by rkmourya** [comment]
I am using Symfony 4.3.
Above setting worked for me but getting deprecation errors.

```
[info] User Deprecated: The "security.authentication.provider.simple_preauth.api" service is deprecated since Symfony 4.2.
[info] User Deprecated: The "Symfony\Component\Security\Core\Authentication\Provider\SimpleAuthenticationProvider" class is deprecated since Symfony 4.2, use Guard instead.
[info] User Deprecated: The "IDCI\Bundle\KeycloakSecurityBundle\Controller\KeycloakController" class extends "Symfony\Bundle\FrameworkBundle\Controller\Controller" that is deprecated since Symfony 4.2, use "Symfony\Bundle\FrameworkBundle\Controller\AbstractController" instead.
[info] User Deprecated: Referencing controllers with IDCIKeycloakSecurityBundle:Keycloak:connect is deprecated since Symfony 4.1, use "IDCI\Bundle\KeycloakSecurityBundle\Controller\KeycloakController::connectAction" instead.
[info] User Deprecated: Referencing controllers with IDCIKeycloakSecurityBundle:Keycloak:connectCheck is deprecated since Symfony 4.1, use "IDCI\Bundle\KeycloakSecurityBundle\Controller\KeycloakController::connectCheckAction" instead.
[info] User Deprecated: Referencing controllers with IDCIKeycloakSecurityBundle:Keycloak:logout is deprecated since Symfony 4.1, use "IDCI\Bundle\KeycloakSecurityBundle\Controller\KeycloakController::logoutAction" instead.
```

**Comment by Brah0um** [comment]
#6

**Comment by Brah0um** [comment]
@rkmourya  The issue is fixed with #6.

Thanks for your report! :+1:
